craft_parts/state_manager/build_state.py

```diff
@@ -25,6 +27,14 @@ class BuildState(StepState):
     """Context information for the build step."""
 
     assets: Dict[str, Any] = {}
+    overlay_hash: Optional[str] = None
+
+    @validator("overlay_hash")
+    @classmethod
+    def _validate_hex_string(cls, value):
```

Comment:
please add a docstring here (and type hints, if you want); thanks!

Reply:
Added docstring.